src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
+            {
+                try
+                {
+                    object result = CachedExpressionCompiler.Process(expression)((TModel)container);
+                    if (expression.ReturnType.GetTypeInfo().IsEnum)
+                    {
+                        var enumValue = metadata.EnumNamesAndValues[result.ToString()];
```

Comment:
I don't think I follow what you mean by
>enum has a value that doesn't match one of its fields

Could you maybe rephrase that or give an expanded example?